types/react-instantsearch-core/index.d.ts

```diff
@@ -748,37 +748,83 @@ export type InsightsClient = (method: InsightsClientMethod, payload: InsightsCli
 export interface ConnectHitInsightsProvided {
-    hit: Hit;
-    insights: WrappedInsightsClient;
+  hit: Hit;
+  insights: WrappedInsightsClient;
 }
 
 export function EXPERIMENTAL_connectConfigureRelatedItems(
   Composed: React.ComponentType<any>
 ): React.ComponentClass<any>;
 export function connectQueryRules(Composed: React.ComponentType<any>): React.ComponentClass<any>;
 export function connectHitInsights(
-    insightsClient: InsightsClient,
+  insightsClient: InsightsClient,
 ): (
-    hitComponent: React.ComponentType<any>,
+  hitComponent: React.ComponentType<any>,
 ) => React.ComponentType<Omit<ConnectHitInsightsProvided, { insights: WrappedInsightsClient }>>;
 export function connectVoiceSearch(Composed: React.ComponentType<any>): React.ComponentClass<any>;
 
-export interface DynamicWidgetsProps  {
-    children: React.ReactNode;
-    attributesToRender: string[];
-    fallbackComponent?: React.ComponentType<{ attribute: string }>;
+export type DynamicWidgetComponent = React.Component<{
+  children?: DynamicWidgetComponent;
+  attribute?: string;
+  attributes?: string[];
+}>;
+
+export interface DynamicWidgetsExposed {
+  /**
+   * The children of this component will be displayed dynamically based
+   * on the result of facetOrdering. This means that any child needs
+   * to have either the “attribute” or “attributes” prop.
+   */
+  children?: React.ReactNode | DynamicWidgetComponent;
+  /**
+   * A function to transform the attributes to render,
+   * or using a different source to determine the attributes to render.
+   */
+  transformItems?: ((...args: any[]) => any) | undefined;
+  /**
+   * The fallbackComponent prop is used if no widget from children matches.
+   * The component gets called with an attribute prop.
+   */
+  fallbackComponent?: React.ComponentType<{ attribute: string }>;
+  /**
+   * The facets to apply before dynamic widgets get mounted.
+   * Setting the value to ['*'] will request all facets
+   * and avoid an additional network request once the widgets are added.
+   * The default value is ['*'].
+   */
+  facets?: string[];
+  /**
+   * The default number of facet values to request.
+   * It’s recommended to have this value at least as high as the highest limit
+   * and showMoreLimit of dynamic widgets, as this will prevent
+   * a second network request once that widget mounts.
+   * To avoid pinned items not showing in the result, make sure you choose
+   * a maxValuesPerFacet at least as high as all the most pinned items you have.
+   * The default value is 20.
+   */
+  maxValuesPerFacet?: number;
 }
 
-export class DynamicWidgets extends React.Component<DynamicWidgetsProps> {}
+export type DynamicWidgetsProvided = Pick<DynamicWidgetsExposed, 'children' | 'fallbackComponent'> & {
+  /** The list of refinement values to display returned from the Algolia API. */
+  attributesToRender: string[]
+};
```

Comment:
I use a type here to extends from `DynamicWidgetsExposed` and pick some types. WDYT?